### PR TITLE
Removing notices for undefined indexes in TODO/FIXME sniffs.

### PR DESCRIPTION
**Pull request by iammichiel**
When a TODO/FIXME message is empty, the sniff raises a notice when trying to access an undefined index of the array.

And adding a second commit removing a NOTICE send on empty function `@throws` comment.

### CODE REVIEW

File: CodeSniffer/Standards/Generic/Sniffs/Commenting/FixmeSniff.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
             // Clear whitespace and some common characters not required at
             // the end of a fixme message to make the error more informative.
             $type         = 'CommentFound';
-            $fixmeMessage = trim($matches[1]);
+            $fixmeMessage = trim(@$matches[1]);
```
Comment (aik099) [inline CodeSniffer/Standards/Generic/Sniffs/Commenting/FixmeSniff.php]: Hiding errors is bad practice especially within a library, that enforces good practice.

I recommend doing proper array key existence check.

File: CodeSniffer/Standards/Generic/Sniffs/Commenting/TodoSniff.php [hunk 2]
```diff
@@ -70,7 +70,7 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
             // Clear whitespace and some common characters not required at
             // the end of a to-do message to make the warning more informative.
             $type        = 'CommentFound';
-            $todoMessage = trim($matches[1]);
+            $todoMessage = trim(@$matches[1]);
```
Comment (aik099) [inline CodeSniffer/Standards/Generic/Sniffs/Commenting/TodoSniff.php]: Hiding errors is bad practice especially within a library, that enforces good practice.

I recommend doing proper array key existence check.

**Comment by iammichiel** [comment]
Did a proper check and validated PHPCS checkstyle.

**Comment by gsherwood** [comment]
I dont get notices for empty FIXME and TODO comments. Can you please provide the sample code you are testing on that produces the notices. I might also need to know the PHP version.

**Comment by iammichiel** [comment]
Well you can reproduce it on any code snippet with empty content. 

```
<?php 

/**
 * @throws
 */
function testFunction() {
    throw new \Exception(); 
    // TODO
    // FIXME
}
```

Tested on two different computers (linux/mac) 
PHP version : 5.5.20
However error, reporting was defined to `E_ALL`

**Comment by iammichiel** [comment]
My bad. When trying to find a working example I actually found the **real** error. 
In fact, the problem is the use of the preg_match expression. As defined in the official PHP documentation : 

> preg_match() returns 1 if the pattern matches given subject, 0 if it does not, or FALSE if an error occurred.

However the condition used in the sniffs is : 

```
 if (preg_match('/(?:\A|[^\p{L}]+)fixme([^\p{L}]+(.*)|\Z)/ui', $content, $matches) !== 0) {
```

Which will in case of FALSE, enter the block code. Which it should not. I will submit a new PR correcting this behaviour.

**Pull request closed without merging**